### PR TITLE
[Ceph] Add Integration Package with Cluster Disk data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (Cluster Disk metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ceph directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #4871
- Relates #4930
- Relates #4963
- Relates #5070 
- Relates #5128 
- Relates #5152 
- Relates #5174

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Comment by agithomas** [comment]
@kush-elastic , please complete the peer review.

@SubhrataK , we have only two small panels in one dashboard. Would you recommend to have one dashboard for all ceph datastreams due to lack of visualisation panels per dashboard

File: packages/ceph/data_stream/cluster_disk/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+- name: ceph
+  type: group
+  fields:
+    - name: cluster_disk
+      type: group
+      fields:
+        - name: available.bytes
```
Comment (agithomas) [inline packages/ceph/data_stream/cluster_disk/fields/fields.yml:7]: So many more unmapped fields are present such as "num_osds, "num_per_pool_osds", "total_used_raw_ratio", "num_per_pool_osds"  etc which i feel are useful. 

@SubhrataK , a more detailed view of storage is available which is not considered in this integration. This view will available when there exist a pool configuration. Also, SSD / HDD based storage classification is available.


Screenshot 2023-01-27 at 11 38 07 AM

Can you please re-check , if we stick to representing only summary or prefer a detailed view  to represent every data that is available as part of running the command via API

File: packages/ceph/data_stream/cluster_disk/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          description: Total bytes of the cluster disk.
+          metric_type: gauge
+          unit: byte
+        - name: used
+          type: group
+          fields:
+            - name: raw.bytes
+              type: long
+              description: Used raw bytes of the cluster disk.
+              metric_type: gauge
```
Comment (agithomas) [inline packages/ceph/data_stream/cluster_disk/fields/fields.yml:23]: Does a used byte have a property of gauge or counter at best?

File: packages/ceph/data_stream/cluster_disk/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+              unit: byte
+            - name: bytes
+              type: long
+              description: Used bytes of the cluster disk.
+              metric_type: gauge
```
Comment (agithomas) [inline packages/ceph/data_stream/cluster_disk/fields/fields.yml:28]: Does a used byte have a property of gauge or counter at best?

Comment (harnish-elastic) [reply packages/ceph/data_stream/cluster_disk/fields/fields.yml:28]: In cluster if the user override/delete the data or the osd are removed/down, the value can be fluctuate. We have performed the same scenarios. Hence it's `gauge`.

File: packages/ceph/data_stream/cluster_disk/fields/fields.yml (hunk 2 above)
Comment (harnish-elastic) [reply packages/ceph/data_stream/cluster_disk/fields/fields.yml:23]: In cluster if the user override/delete the data or the osd are removed/down, the value can be fluctuate. We have performed the same scenarios. Hence it's `gauge`.

**Review by agithomas — APPROVED** [review]
LGTM

File: packages/ceph/_dev/build/docs/README.md [hunk 4]
```diff
@@ -6,17 +6,18 @@
...
 
 ## Data streams
 
 The Ceph integration collects metrics data.
 
-Metrics give you insight into the statistics of the Ceph. The Metric data streams collected by the Ceph integration are `cluster_health`, `osd_performance`, `osd_pool_stats` and `osd_tree`, so that the user can monitor and troubleshoot the performance of the Ceph instance.
+Metrics give you insight into the statistics of the Ceph. The Metric data streams collected by the Ceph integration are `cluster_disk`, `cluster_health`, `osd_performance`, `osd_pool_stats` and `osd_tree`, so that the user can monitor and troubleshoot the performance of the Ceph instance.
 
 Data stream:
+- `cluster_disk`: Tracks overall storage of the cluster.
```
Comment (agithomas) [inline packages/ceph/_dev/build/docs/README.md:20]: The word track and Represents are used without consistency. Suggest to use words  <dataset name> collects / fetches.

File: packages/ceph/docs/README.md [hunk 5]
```diff
@@ -6,17 +6,18 @@
...
 
 ## Data streams
 
 The Ceph integration collects metrics data.
 
-Metrics give you insight into the statistics of the Ceph. The Metric data streams collected by the Ceph integration are `cluster_health`, `osd_performance`, `osd_pool_stats` and `osd_tree`, so that the user can monitor and troubleshoot the performance of the Ceph instance.
+Metrics give you insight into the statistics of the Ceph. The Metric data streams collected by the Ceph integration are `cluster_disk`, `cluster_health`, `osd_performance`, `osd_pool_stats` and `osd_tree`, so that the user can monitor and troubleshoot the performance of the Ceph instance.
 
 Data stream:
+- `cluster_disk`: Tracks overall storage of the cluster.
```
Comment (agithomas) [inline packages/ceph/docs/README.md:20]: Same with the above docs comment. Please use <data_stream name> collects / fetches

**Comment by agithomas** [comment]
LGTM

**Review by agithomas — APPROVED** [review]
Approved

**Pull request merged**